server/src/main/java/org/elasticsearch/index/mapper/TypeFieldMapper.java

```diff
@@ -124,6 +125,7 @@ public boolean isSearchable() {
 
         @Override
         public Query existsQuery(QueryShardContext context) {
```

Comment:
To be really comprehensive, we could also override the other query types (`fuzzy`, `prefix`, `wildcard` and `regex`) and issue deprecation warnings.

Reply:
Thanks Julie. Turns out that `fuzzy`, `prefix` and `regex` query all fail on the _type field, as the field is not indexed. So these kind of requests are not allowed. `wildcard` query is calling `termQuery` underneath, so we should get an expected deprecation message.